src/ipv6/ipv6_wire.ml

```diff
@@ -103,6 +103,15 @@ let get_opt_prefix_on_link buf =
 let get_opt_prefix_autonomous buf =
   get_opt_prefix_reserved1 buf land 0x40 <> 0
 
+[%%cstruct
+type opt_rdnss = {
+    ty:                 uint8_t;
+    len:                uint8_t;
+    rdnss_lifetime:     uint32_t;
+    rdnss_addresses:    uint8_t [@len 16];
```

Comment:
keep in mind that the addresses are any number of addresses (actually "len - 1" ones) -- but I can't see a way in ppx_cstruct to define "now there's a list of 16 byte values, in the number of len - 1". so it is fine to leave this as is (maybe add a comment), but the actualy decoding and encoding part needs to be aware of that.

Reply:
so maybe just skip the `rdnss_addresses` entirely here (and have the `type opt_rdnss_header`)?